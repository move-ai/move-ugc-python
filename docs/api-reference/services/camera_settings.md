# Camera settings

::: move_ugc.services.camera_settings.CameraSettingsService
    options:
        members:
            - list
