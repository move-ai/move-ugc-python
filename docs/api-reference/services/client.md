# Client

::: move_ugc.services.client.ClientService
    options:
        members:
            - retrieve
            - update