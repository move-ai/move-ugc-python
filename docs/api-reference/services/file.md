# Files

::: move_ugc.services.file.FileService
    options:
        members:
            - create
            - retrieve