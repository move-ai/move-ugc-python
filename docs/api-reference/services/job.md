# Jobs

::: move_ugc.services.job.JobService
    options:
        members:
            - create
            - retrieve
            - update
            - list
