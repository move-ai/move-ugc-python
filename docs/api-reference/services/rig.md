# Rig

::: move_ugc.services.rig.RigService
    options:
        members:
            - list
