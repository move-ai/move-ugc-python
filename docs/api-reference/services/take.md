# Takes

::: move_ugc.services.take.TakeService
    options:
        members:
            - create_singlecam
            - create_multicam
            - retrieve
            - list
            - update