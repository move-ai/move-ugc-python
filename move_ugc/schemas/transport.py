"""Define gql transports for the client."""
from gql import Client
from gql.transport.requests import RequestsHTTPTransport
from pydantic import HttpUrl

from move_ugc.schemas.constants import AUTHORIZATION_HEADER


def _get_transport(endpoint_url: HttpUrl, api_key: str) -> RequestsHTTPTransport:
    """Get the transport for the GraphQL client.

    Args:
        endpoint_url: The endpoint URL to use for the transport.
        api_key: The API key to use for the transport.

    Returns:
        The transport for the GraphQL client.
    """
    return RequestsHTTPTransport(
        url=str(endpoint_url),
        use_json=True,
        headers={AUTHORIZATION_HEADER: api_key},
        verify=True,
    )


def _get_gql_client(transport: RequestsHTTPTransport) -> Client:
    """Get the GraphQL client.

    Args:
        transport: The transport to use for the client.

    Returns:
        The GraphQL client.
    """
    return Client(transport=transport, fetch_schema_from_transport=True)
