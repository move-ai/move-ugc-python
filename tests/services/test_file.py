"""Unit tests for using the files service."""
import pytest
from gql.transport.exceptions import TransportQueryError

from tests.services.testcases import ServicesTestCase


class TestFileService(ServicesTestCase):
    """Test file service."""

    service_name = "files"

    @pytest.mark.parametrize(
        argnames="expand",
        argvalues=[None, [], ["client"]],
        ids=["no_expand", "empty_expand", "expand_client"],
    )
    def test_retrieve(self, expand, snapshot, faker, file_retrieve_response):
        """Test retrieving a file from Move UGC API.

        This should test -> `ugc.files.retrieve(id='files-<uuid>)'`

        Args:
            expand: The expand fixture.
            snapshot: The snapshot fixture.
            faker: The faker fixture.
            file_retrieve_response: The file retrieve response fixture.
        """
        file_model = self.client.files.retrieve(id=faker.uuid4(), expand=expand)
        suffix = "_".join(expand or [])
        self.assert_execute(
            snapshot,
            name=f"retrieve_query_expand_{suffix}",
        )
        snapshot.assert_match(
            file_model.model_dump(),
            name=f"file_retrieve_response_expand_{suffix}",
        )

    def test_file_not_found(self, snapshot, faker, file_not_found_response):
        """Test file not found.

        This should test -> `ugc.files.retrieve(id='invalid-id')'`

        Args:
            snapshot: The snapshot fixture.
            faker: The faker fixture.
            file_not_found_response: The file not found response fixture.
        """
        with pytest.raises(TransportQueryError) as excinfo:
            self.client.files.retrieve(id=faker.uuid4(), expand=["client"])
        snapshot.assert_match(
            excinfo.value.errors,  # noqa: WPS441
            name="file_not_found_response",
        )
