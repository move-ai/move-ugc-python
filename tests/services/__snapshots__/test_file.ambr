# serializer version: 1
# name: TestFileService.test_create[empty_expand][create_mutation_expand_[]]
  list([
    list([
      '''
        mutation create($type: String!, $metadata: AWSJSON, $name: String) {
          createFile(type: $type, metadata: $metadata, name: $name) {
            id
            created
            type
            metadata
            name
            presignedUrl
            thumbnailUrl
            __typename
          }
        }
      ''',
    ]),
    dict({
      'operation_name': None,
      'variable_values': dict({
        'metadata': '{"decade": "zAvJMvacZIYSmMsDUNvC", "other": "XbBPNrbhtJksbBuoWXSK", "draw": "PnbQcVNCliYtuFCSJkGb", "table": "ACMVZcKAiGBcYgCzHAad", "huge": "eFaLyHGEQQpSzpRFSYEm", "last": "dnZCcfgZNBnaEkbOzIyO", "trouble": "UgnhIyaDJzohUigyDYZf", "analysis": "UmKdTFlLMIuIvJkRJnoM", "house": "aYyOUXkJPUjPJGpDdakX", "director": "KNmpExWtgQLcAEuRyBkN"}',
        'name': '',
        'type': 'png',
      }),
    }),
  ])
# ---
# name: TestFileService.test_create[expand_client][create_mutation_expand_client]
  list([
    list([
      '''
        mutation create($type: String!, $metadata: AWSJSON, $name: String) {
          createFile(type: $type, metadata: $metadata, name: $name) {
            id
            created
            type
            metadata
            name
            presignedUrl
            thumbnailUrl
            __typename
            client {
              id
              name
              created
              metadata
              portal
              __typename
            }
          }
        }
      ''',
    ]),
    dict({
      'operation_name': None,
      'variable_values': dict({
        'metadata': '{"decade": "zAvJMvacZIYSmMsDUNvC", "other": "XbBPNrbhtJksbBuoWXSK", "draw": "PnbQcVNCliYtuFCSJkGb", "table": "ACMVZcKAiGBcYgCzHAad", "huge": "eFaLyHGEQQpSzpRFSYEm", "last": "dnZCcfgZNBnaEkbOzIyO", "trouble": "UgnhIyaDJzohUigyDYZf", "analysis": "UmKdTFlLMIuIvJkRJnoM", "house": "aYyOUXkJPUjPJGpDdakX", "director": "KNmpExWtgQLcAEuRyBkN"}',
        'name': '',
        'type': 'png',
      }),
    }),
  ])
# ---
# name: TestFileService.test_create[no_expand][create_mutation_expand_None]
  list([
    list([
      '''
        mutation create($type: String!, $metadata: AWSJSON, $name: String) {
          createFile(type: $type, metadata: $metadata, name: $name) {
            id
            created
            type
            metadata
            name
            presignedUrl
            thumbnailUrl
            __typename
          }
        }
      ''',
    ]),
    dict({
      'operation_name': None,
      'variable_values': dict({
        'metadata': '{"decade": "zAvJMvacZIYSmMsDUNvC", "other": "XbBPNrbhtJksbBuoWXSK", "draw": "PnbQcVNCliYtuFCSJkGb", "table": "ACMVZcKAiGBcYgCzHAad", "huge": "eFaLyHGEQQpSzpRFSYEm", "last": "dnZCcfgZNBnaEkbOzIyO", "trouble": "UgnhIyaDJzohUigyDYZf", "analysis": "UmKdTFlLMIuIvJkRJnoM", "house": "aYyOUXkJPUjPJGpDdakX", "director": "KNmpExWtgQLcAEuRyBkN"}',
        'name': '',
        'type': 'png',
      }),
    }),
  ])
# ---
# name: TestFileService.test_fetch_service
  FileService(api_key=SecretStr('**********'), endpoint_url=HttpUrl('https://pytest_invalid_endpoint_url.com/'))
# ---
# name: TestFileService.test_file_not_found[file_not_found_response]
  list([
    dict({
      'data': None,
      'errorInfo': None,
      'errorType': 'MoveNotFoundError',
      'locations': list([
        dict({
          'column': 3,
          'line': 2,
          'sourceName': None,
        }),
      ]),
      'message': 'Resource not found in move-ugc-api.',
      'path': list([
        'getFile',
      ]),
    }),
  ])
# ---
# name: TestFileService.test_generate_share_code[generate_share_code_mutation]
  list([
    list([
      '''
        mutation generateShareCode($fileId: String!) {
          generateShareCode(fileId: $fileId) {
            code
            created
            file {
              id
            }
            expires
            url
            __typename
          }
        }
      ''',
    ]),
    dict({
      'operation_name': None,
      'variable_values': dict({
        'fileId': '516b93c2-ca3c-4b89-b7d8-6704858462ab',
      }),
    }),
  ])
# ---
# name: TestFileService.test_retrieve[empty_expand][retrieve_query_expand_[]]
  list([
    list([
      '''
        query retrieve($id: ID!) {
          getFile(fileId: $id) {
            id
            created
            type
            metadata
            name
            presignedUrl
            thumbnailUrl
            __typename
          }
        }
      ''',
    ]),
    dict({
      'operation_name': None,
      'variable_values': dict({
        'id': '8d866324-3ab5-4d05-a9c5-e560a8020f0a',
      }),
    }),
  ])
# ---
# name: TestFileService.test_retrieve[empty_expand_no_thumbnail][retrieve_query_expand_[]]
  list([
    list([
      '''
        query retrieve($id: ID!) {
          getFile(fileId: $id) {
            id
            created
            type
            metadata
            name
            presignedUrl
            thumbnailUrl
            __typename
          }
        }
      ''',
    ]),
    dict({
      'operation_name': None,
      'variable_values': dict({
        'id': '8d866324-3ab5-4d05-a9c5-e560a8020f0a',
      }),
    }),
  ])
# ---
# name: TestFileService.test_retrieve[expand_client][retrieve_query_expand_client]
  list([
    list([
      '''
        query retrieve($id: ID!) {
          getFile(fileId: $id) {
            id
            created
            type
            metadata
            name
            presignedUrl
            thumbnailUrl
            __typename
            client {
              id
              name
              created
              metadata
              portal
              __typename
            }
          }
        }
      ''',
    ]),
    dict({
      'operation_name': None,
      'variable_values': dict({
        'id': '8d866324-3ab5-4d05-a9c5-e560a8020f0a',
      }),
    }),
  ])
# ---
# name: TestFileService.test_retrieve[no_expand][retrieve_query_expand_None]
  list([
    list([
      '''
        query retrieve($id: ID!) {
          getFile(fileId: $id) {
            id
            created
            type
            metadata
            name
            presignedUrl
            thumbnailUrl
            __typename
          }
        }
      ''',
    ]),
    dict({
      'operation_name': None,
      'variable_values': dict({
        'id': '8d866324-3ab5-4d05-a9c5-e560a8020f0a',
      }),
    }),
  ])
# ---
# name: TestFileService.test_update[empty_expand][update_mutation_expand_[]]
  list([
    list([
      '''
        mutation update($id: String!, $metadata: AWSJSON, $name: String) {
          updateFile(id: $id, metadata: $metadata, name: $name) {
            id
            created
            type
            metadata
            name
            presignedUrl
            thumbnailUrl
            __typename
          }
        }
      ''',
    ]),
    dict({
      'operation_name': None,
      'variable_values': dict({
        'id': '8d866324-3ab5-4d05-a9c5-e560a8020f0a',
        'metadata': '{"decade": "zAvJMvacZIYSmMsDUNvC", "other": "XbBPNrbhtJksbBuoWXSK", "draw": "PnbQcVNCliYtuFCSJkGb", "table": "ACMVZcKAiGBcYgCzHAad", "huge": "eFaLyHGEQQpSzpRFSYEm", "last": "dnZCcfgZNBnaEkbOzIyO", "trouble": "UgnhIyaDJzohUigyDYZf", "analysis": "UmKdTFlLMIuIvJkRJnoM", "house": "aYyOUXkJPUjPJGpDdakX", "director": "KNmpExWtgQLcAEuRyBkN"}',
        'name': 'Amber Harris',
      }),
    }),
  ])
# ---
# name: TestFileService.test_update[expand_client][update_mutation_expand_client]
  list([
    list([
      '''
        mutation update($id: String!, $metadata: AWSJSON, $name: String) {
          updateFile(id: $id, metadata: $metadata, name: $name) {
            id
            created
            type
            metadata
            name
            presignedUrl
            thumbnailUrl
            __typename
            client {
              id
              name
              created
              metadata
              portal
              __typename
            }
          }
        }
      ''',
    ]),
    dict({
      'operation_name': None,
      'variable_values': dict({
        'id': '8d866324-3ab5-4d05-a9c5-e560a8020f0a',
        'metadata': '{"decade": "zAvJMvacZIYSmMsDUNvC", "other": "XbBPNrbhtJksbBuoWXSK", "draw": "PnbQcVNCliYtuFCSJkGb", "table": "ACMVZcKAiGBcYgCzHAad", "huge": "eFaLyHGEQQpSzpRFSYEm", "last": "dnZCcfgZNBnaEkbOzIyO", "trouble": "UgnhIyaDJzohUigyDYZf", "analysis": "UmKdTFlLMIuIvJkRJnoM", "house": "aYyOUXkJPUjPJGpDdakX", "director": "KNmpExWtgQLcAEuRyBkN"}',
        'name': 'Amber Harris',
      }),
    }),
  ])
# ---
# name: TestFileService.test_update[no_expand][update_mutation_expand_None]
  list([
    list([
      '''
        mutation update($id: String!, $metadata: AWSJSON, $name: String) {
          updateFile(id: $id, metadata: $metadata, name: $name) {
            id
            created
            type
            metadata
            name
            presignedUrl
            thumbnailUrl
            __typename
          }
        }
      ''',
    ]),
    dict({
      'operation_name': None,
      'variable_values': dict({
        'id': '8d866324-3ab5-4d05-a9c5-e560a8020f0a',
        'metadata': '{"decade": "zAvJMvacZIYSmMsDUNvC", "other": "XbBPNrbhtJksbBuoWXSK", "draw": "PnbQcVNCliYtuFCSJkGb", "table": "ACMVZcKAiGBcYgCzHAad", "huge": "eFaLyHGEQQpSzpRFSYEm", "last": "dnZCcfgZNBnaEkbOzIyO", "trouble": "UgnhIyaDJzohUigyDYZf", "analysis": "UmKdTFlLMIuIvJkRJnoM", "house": "aYyOUXkJPUjPJGpDdakX", "director": "KNmpExWtgQLcAEuRyBkN"}',
        'name': 'Amber Harris',
      }),
    }),
  ])
# ---
